engine/order_manager.go

```diff
@@ -589,6 +589,14 @@ func (m *OrderManager) processSubmittedOrder(newOrderResp *order.SubmitResponse)
 		return nil, err
 	}
 
+	if err := m.orderStore.add(detail.CopyToPointer()); errors.Is(err, ErrOrdersAlreadyExists) {
+		// Streamed by ws before we got here. Details from ws supecede since they are more recent.
+		detail = m.orderStore.getByDetail(detail)
```

Comment:
This is mostly a comment on the "superseded" part and I'm not requesting a change.
This is a little risky, but for the most part its true. For ZB and Kraken however, we actually lose detail by discarding the REST response. However, I will rely on `processOrders` to ensure orders are up to date and not request a change.

Reply:
I see you've learnt not to bait me with challenges :joy: